src/libraries/System.Linq.Queryable/src/System/Linq/TypeHelper.cs

```diff
@@ -3,13 +3,14 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Reflection;
 
 namespace System.Linq
 {
     internal static class TypeHelper
     {
-        internal static Type FindGenericType(Type definition, Type type)
+        internal static Type? FindGenericType(Type definition, [NotNull]Type? type)
```

Comment:
Doesn't really understood why we need `[NotNull]`attribute

Reply:
The contract for the `type` argument is not null, however the method body will assign null values to it. The attribute compromises the two without having to declare a new local variable.

Reply:
I understand the logic, but I suggest removing the [NotNull].  The implementation works fine if null is passed in, this is internal so there's no documented contract it needs to expose, and the combination of [NotNull] and ? just looks strange and leads to questions.